vllm/model_executor/layers/quantization/utils/fp8_utils.py

```diff
@@ -98,122 +101,226 @@ def rocm_aiter_gemm_w8a8_blockscale_fake(
                                                      block_size[1],
-                                                     column_major_scales=True)
-        output = w8a8_blockscale_func(q_input, weight, x_scale, weight_scale,
-                                      block_size, input.dtype)
-        if num_pad > 0:
-            output = output[:-num_pad]
-    else:
-        if use_aiter_and_is_supported:
-            q_input, x_scale = aiter_per1x128_quant(
-                input_2d.contiguous(), quant_dtype=rocm_aiter.dtypes.fp8)
+                                                     column_major_scales=True,
+                                                     use_ue8m0=False)
+        if self.is_hopper:
+            output = torch.ops.vllm.padded_cutlass_scaled_mm(
+                q_input, weight, x_scale, weight_scale, block_size,
+                input_2d.dtype)
         else:
-            q_input, x_scale = per_token_group_quant_fp8(
-                input_2d, block_size[1], column_major_scales=False)
-
-        output = w8a8_blockscale_func(q_input, weight, x_scale, weight_scale,
-                                      block_size, input.dtype)
-
-    if bias is not None:
-        output = output + bias
-    return output.to(dtype=input.dtype).view(*output_shape)
-
-
-def apply_w8a8_block_fp8_linear_fake(
-    input: torch.Tensor,
-    weight: torch.Tensor,
-    block_size: list[int],
-    weight_scale: torch.Tensor,
-    input_scale: Optional[torch.Tensor] = None,
-    bias: Optional[torch.Tensor] = None,
-    cutlass_block_fp8_supported: bool = CUTLASS_BLOCK_FP8_SUPPORTED,
-    use_aiter_and_is_supported: bool = False,
-) -> torch.Tensor:
-    output_shape = [*input.shape[:-1], weight.shape[0]]
-    return torch.empty(output_shape, dtype=input.dtype, device=input.device)
-
+            output = cutlass_scaled_mm(q_input, weight, x_scale, weight_scale,
+                                       block_size, input_2d.dtype, False)
+        return output
 
-if not current_platform.is_cpu():
-    direct_register_custom_op(
-        op_name="apply_w8a8_block_fp8_linear",
-        op_func=apply_w8a8_block_fp8_linear,
-        mutates_args=[],
-        fake_impl=apply_w8a8_block_fp8_linear_fake,
-    )
+    def _run_aiter(
+        self,
+        input_2d: torch.Tensor,
+        weight: torch.Tensor,
+        block_size: list[int],
+        weight_scale: torch.Tensor,
+    ) -> torch.Tensor:
+        q_input, x_scale = aiter_per1x128_quant(
+            input_2d.contiguous(), quant_dtype=rocm_aiter.dtypes.fp8)
+        return torch.ops.vllm.rocm_aiter_gemm_w8a8_blockscale(
+            q_input, weight, x_scale, weight_scale, block_size, input_2d.dtype)
+
+    def _run_w8a8_block_fp8_matmul(
+        self,
+        input_2d: torch.Tensor,
+        weight: torch.Tensor,
+        block_size: list[int],
+        weight_scale: torch.Tensor,
+    ) -> torch.Tensor:
+        q_input, x_scale = per_token_group_quant_fp8(input_2d,
```

Comment:
Can we use the QuantFP8 abstraction here, and in config init we can enable the custom kernel (so that `per_token_group_quant_fp8` runs) except on Blackwell (and link issue: #25094).

Reply:
Is `forward_cuda()` also undeperforming? From what I'm seeing, it's what would replace `per_token_group_quant_fp8()`

Reply:
In any case, I needed to modify QuantFP8 with one extra arg for compilation (so it won't complain when we call forward)

Reply:
When enabled, `QuantFP8` will use the `per_token_group_quant_fp8` kernel, otherwise it will use the `forward_native` torch implementation.